### ISSUE TITLE
Deserializing an `Option<Vec<_>>`

### ISSUE THREAD

**Issue by kardeiz** [samscott89/serde_qs#2]
It looks like `serde_qs` can currently only deserialize options where `Option<String>` (it seems like it can serialize any `Option<_>` just fine). I have a case where a field is `Option<Vec<String>>`, and I'm getting the error `"value does not appear to be a value"`.

I resolved this by going into `src/de.rs` and adding the following lines to `LevelDeserializer::deserialize_option` (inside the pattern match):

```
Level::Sequence(x) => {
    visitor.visit_some(SeqDeserializer::new(x.into_iter()))
},
Level::Nested(map) => {
    visitor.visit_some(SeqDeserializer::new(map.into_iter().map(|(_k, v)| v)))
},
```

This seems to be working fine for my use. I haven't opened a PR because I have no idea if this is a legitimate way to deserialize an `Option<Vec<_>>` or if it would break anything elsewhere.

**Comment by samscott89** [comment]
Does commit https://github.com/samscott89/serde_qs/commit/81b77bd7e7f3397ddfeeb9a63f5c74a08fd6837c do the trick?


Similar idea to what you wrote, but lets `deserialize` dispatch the various deserializers more generically, so that `visitor.visit_some(self)` works for values, maps and sequences.

Thanks for raising the issue.

**Comment by kardeiz** [comment]
Works perfect, thanks so much!

**Comment by samscott89** [comment]
No worries. I've published it as `0.0.2` if you want to get it from crates.io.
